addons/l10n_ec/migrations/3.4/post-migration.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, SUPERUSER_ID
+
+def update_withhold_type(env):
+    # Recreate account tax group templates
+    ChartTemplate = env['account.chart.template']
+    tax_group_data = ChartTemplate._get_account_tax_group('ec')
+    for company in env['res.company'].search([('chart_template', '=', 'ec')]):
+        ChartTemplate = env['account.chart.template'].with_company(company)
+        ChartTemplate._load_data({'account.tax.group': tax_group_data})
+
+    # Reclassifies withhold taxes into independent tax groups for sales and purchases
+    env.cr.execute('''
+        UPDATE account_tax t
+        SET tax_group_id=tgnew.id 
+        FROM account_tax_group tgnew, account_tax_group tgold
+        WHERE t.type_tax_use='sale'
+            AND tgold.id = t.tax_group_id
+            AND tgold.l10n_ec_type='withhold_vat'
+            AND tgnew.company_id = t.company_id
+            AND tgnew.l10n_ec_type='withhold_vat_sale'
+    ''')
+    env.cr.execute('''
+        UPDATE account_tax t
+        SET tax_group_id=tgnew.id 
+        FROM account_tax_group tgnew, account_tax_group tgold
+        WHERE t.type_tax_use='purchase'
+            AND tgold.id = t.tax_group_id
+            AND tgold.l10n_ec_type='withhold_vat'
+            AND tgnew.company_id = t.company_id
+            AND tgnew.l10n_ec_type='withhold_vat_purchase'
+    ''')
+    env.cr.execute('''
+        UPDATE account_tax t
+        SET tax_group_id=tgnew.id 
+        FROM account_tax_group tgnew, account_tax_group tgold
+        WHERE t.type_tax_use='sale'
+            AND tgold.id = t.tax_group_id
+            AND tgold.l10n_ec_type='withhold_income_tax'
+            AND tgnew.company_id = t.company_id
+            AND tgnew.l10n_ec_type='withhold_income_sale'
+    ''')
+    env.cr.execute('''
+        UPDATE account_tax t
+        SET tax_group_id=tgnew.id 
+        FROM account_tax_group tgnew, account_tax_group tgold
+        WHERE t.type_tax_use='purchase'
+            AND tgold.id = t.tax_group_id
+            AND tgold.l10n_ec_type='withhold_income_tax'
+            AND tgnew.company_id = t.company_id
+            AND tgnew.l10n_ec_type='withhold_income_purchase'
+    ''')
+
+def update_type_tax_use(env):
+    # sets type_tax_use = none for withholding taxes
+    env.cr.execute('''
+        UPDATE account_tax
+        SET type_tax_use = 'none'
+        WHERE tax_group_id IN (SELECT id FROM account_tax_group WHERE l10n_ec_type IN ('withhold_income_purchase','withhold_vat_purchase','withhold_income_sale','withhold_vat_sale'))
+    ''')
+
+def update_tax_repartition_line_vat_withhold(env):
+    # For tax repartition lines in vat withhold taxes, replace factor_percent=12% with factor_percent=100%
+    env.cr.execute('''
+        --for invoice_tax_id
```

Comment:
Not sure about the purpose of this comment

Reply:
idd, only a tax_id now